cmd/juju/status/history.go

```diff
@@ -209,19 +239,37 @@ func (c *statusHistoryCommand) Run(ctx *cmd.Context) error {
 		return errors.Errorf("no status history available")
 	}
 
-	c.writeTabular(ctx.Stdout, statuses)
+	history := make(History, len(statuses))
+	for i, h := range statuses {
+		history[i] = DetailedStatus{
+			Status:  h.Status,
+			Message: h.Info,
+			Data:    h.Data,
+			Since:   h.Since,
+			Kind:    h.Kind,
+		}
+	}
+	return c.out.Write(ctx, history)
+}
+
+func (c *statusHistoryCommand) formatTabular(writer io.Writer, value interface{}) error {
+	h, ok := value.(History)
+	if !ok {
+		return errors.Errorf("expected value of type %T, got %T", History{}, value)
```

Comment:
Any reason not just `errors.Errorf("expected value of type History, got %T", value)`?

Reply:
Not really, just following the template used in other places. I sorta think this is more "pure".